chia/util/file_keyring.py

```diff
@@ -468,8 +469,4 @@ def set_passphrase_hint(self, passphrase_hint: Optional[str]) -> None:
         Store the new passphrase hint in the staging dict (outer_payload_properties_for_next_write) to
         be written-out on the next write to the keyring.
         """
-        assert self.outer_payload_properties_for_next_write is not None
-        if passphrase_hint is not None and len(passphrase_hint) > 0:
-            self.outer_payload_properties_for_next_write["passphrase_hint"] = passphrase_hint
-        elif "passphrase_hint" in self.outer_payload_properties_for_next_write:
-            del self.outer_payload_properties_for_next_write["passphrase_hint"]
+        self.outer_payload_properties_for_next_write["passphrase_hint"] = passphrase_hint
```

Comment:
Correct this in combination with 1114a55b1b6b358b6c2fae0a77d5a8f3b4fba6e1 sets the `passphrase_hint` to `None` in the file if there is no hint set. The `del` expression did only remove from `outer_payload_properties_for_next_write` not from `outer_payload_cache` thats why the removal was broken also. Sure we could add another `if "passphrase_hint" in self.outer_payload_cache: del ...` but then, if the keyring gets reloaded between the call of `set_passphrase_hint(None)` and the next write it just gets reloaded into the cache  and we end up not removing it. We could also drop it before writing based on it being `None` in `outer_payload_properties_for_next_write` but i anyway have another PR coming which leads to all fields always being present in the file so i think we can just leave it like this unless there are any other objections.

Reply:
Ok that sounds fine. I was thinking it would be nice to prune `passphrase_hint` from keyring.yaml if the hint was removed, but setting it to None should be fine.